File: cepaslib/src/main/java/com/itachi1706/cepaslib/persist/CardKeysPersister.java
package com.itachi1706.cepaslib.persist;

import androidx.annotation.NonNull;
import androidx.annotation.Nullable;

import com.itachi1706.cepaslib.persist.db.model.SavedKey;

import java.util.List;

public interface CardKeysPersister {

    @NonNull
    List<SavedKey> getSavedKeys();

    @Nullable
    SavedKey getForTagId(@NonNull String tagId);

    long insert(@NonNull SavedKey savedKey);

    void delete(@NonNull SavedKey savedKey);
}

File: cepaslib/src/main/java/com/itachi1706/cepaslib/persist/CardKeysPersister.kt
package com.itachi1706.cepaslib.persist

import com.itachi1706.cepaslib.persist.db.model.SavedKey

interface CardKeysPersister {
    val savedKeys: List<SavedKey?>
    fun getForTagId(tagId: String): SavedKey?
    fun insert(savedKey: SavedKey): Long
    fun delete(savedKey: SavedKey)
}